modules/imgcodecs/src/apngframe.hpp

```diff
@@ -0,0 +1,173 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level
+// directory of this distribution and at http://opencv.org/license.html
+
+/****************************************************************************\
+ *
+ *  this file includes some modified part of apngasm
+ *
+ ****************************************************************************/
+
+
+ /*  apngasm
+ *
+ *  The next generation of apngasm, the APNG Assembler.
+ *  The apngasm CLI tool and library can assemble and disassemble APNG image files.
+ *
+ *  https://github.com/apngasm/apngasm
+ *
+ * zlib license
+ * ------------
+ *
+ * This software is provided 'as-is', without any express or implied
+ * warranty.  In no event will the authors be held liable for any damages
+ * arising from the use of this software.
+ *
+ * Permission is granted to anyone to use this software for any purpose,
+ * including commercial applications, and to alter it and redistribute it
+ * freely, subject to the following restrictions:
+ *
+ * 1. The origin of this software must not be misrepresented; you must not
+ *    claim that you wrote the original software. If you use this software
+ *    in a product, an acknowledgment in the product documentation would be
+ *    appreciated but is not required.
+ * 2. Altered source versions must be plainly marked as such, and must not be
+ *    misrepresented as being the original software.
+ * 3. This notice may not be removed or altered from any source distribution.
+ *
+ */
+
+#ifndef _APNGFRAME_H_
+#define _APNGFRAME_H_
+
+#include <string>
+
+namespace cv {
+
+const unsigned DEFAULT_FRAME_NUMERATOR =
+    100; //!< @brief The default numerator for the frame delay fraction.
+const unsigned DEFAULT_FRAME_DENOMINATOR =
+    1000; //!< @brief The default denominator for the frame delay fraction.
+
+typedef struct {
+  unsigned char r, g, b;
+} rgb;
+typedef struct {
+  unsigned char r, g, b, a;
+} rgba;
+
+// Individual APNG frame
+class APNGFrame {
+public:
+  // Raw pixel data
+  unsigned char *pixels(unsigned char *setPixels = NULL);
+  unsigned char *_pixels;
+
+  // Width and Height
+  unsigned int width(unsigned int setWidth = 0);
+  unsigned int height(unsigned int setHeight = 0);
+  unsigned int _width;
+  unsigned int _height;
+
+  // PNG color type
+  unsigned char colorType(unsigned char setColorType = 255);
+  unsigned char _colorType;
+
+  // Palette into
+  rgb *palette(rgb *setPalette = NULL);
+  rgb _palette[256];
+
+  // Transparency info
+  unsigned char *transparency(unsigned char *setTransparency = NULL);
+  unsigned char _transparency[256];
+
+  // Sizes for palette and transparency records
+  int paletteSize(int setPaletteSize = 0);
```

Comment:
it's hard to get this is a setter. Call it `setPaletteSize`.
But if _palette_size is public, what's the point of having a setter? Just make the member public no?